include/vcpkg/base/messages.h

```diff
@@ -0,0 +1,259 @@
+#pragma once
+
+#include <vcpkg/base/fwd/files.h>
+#include <vcpkg/base/fwd/json.h>
+#include <vcpkg/base/fwd/lineinfo.h>
+
+//#include <vcpkg/base/stringliteral.h>
+
+//#include <string>
+
+//#include <fmt/format.h>
+
+namespace vcpkg
+{
+#if defined(_WIN32)
+    enum class Color : unsigned short
+    {
+        None = 0,
+        Success = 0x0A, // FOREGROUND_GREEN | FOREGROUND_INTENSITY
+        Error = 0xC,    // FOREGROUND_RED | FOREGROUND_INTENSITY
+        Warning = 0xE,  // FOREGROUND_GREEN | FOREGROUND_RED | FOREGROUND_INTENSITY
+    };
+#else
+    enum class Color : char
+    {
+        None = 0,
+        Success = '2', // [with 9] bright green
+        Error = '1',   // [with 9] bright red
+        Warning = '3', // [with 9] bright yellow
+    };
+#endif
+}
+
+namespace vcpkg::msg
+{
+    struct LocalizedString;
+    struct LocalizedStringView;
+
+    namespace detail
+    {
+        template<class Tag, class Type>
+        struct MessageArgument
+        {
+            const Type* parameter; // always valid
+        };
+
+        template<class... Tags>
+        struct MessageCheckFormatArgs
+        {
+            static constexpr void check_format_args(const Tags&...) noexcept
+            {
+            }
+        };
+
+        LocalizedString internal_vformat(int index, fmt::format_args args);
+
+        template<class... Args>
+        MessageCheckFormatArgs<Args...> make_message_check_format_args(const Args&... args);
+
+        ::size_t startup_register_message(StringLiteral name, StringLiteral format_string, StringLiteral comment);
+
+        ::size_t number_of_messages();
+
+        // REQUIRES: index < last_message_index()
+        StringView get_format_string(::size_t index);
+        // REQUIRES: index < last_message_index()
+        StringView get_message_name(::size_t index);
+        // REQUIRES: index < last_message_index()
+        StringView get_default_format_string(::size_t index);
+        // REQUIRES: index < last_message_index()
+        StringView get_localization_comment(::size_t index);
+    }
+
+    // load from "locale_base/${language}.json"
+    void threadunsafe_initialize_context(const Filesystem& fs, StringView language, const Path& locale_base);
```

Comment:
Noting that the user could set language to `C:\Users\My User\foo` and get that loaded as a language file.